src/OpenTelemetry.Exporter.Prometheus.HttpListener/PrometheusHttpListener.cs

```diff
@@ -187,4 +193,36 @@ private async Task ProcessRequestAsync(HttpListenerContext context)
         {
         }
     }
+
+    private bool AcceptsOpenMetrics(HttpListenerRequest request)
+    {
+        if (request.AcceptTypes == null)
+        {
+            return false;
+        }
+
+        foreach (var acceptType in request.AcceptTypes)
+        {
+            if (this.GetMediaType(acceptType) == OpenMetricsMediaType)
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private string GetMediaType(string acceptHeader)
+    {
+        if (string.IsNullOrEmpty(acceptHeader))
+        {
+            return string.Empty;
+        }
+
+        var separatorIndex = acceptHeader.IndexOf(';');
+
+        return separatorIndex == -1
+            ? acceptHeader
+            : acceptHeader.Substring(0, separatorIndex);
```

Comment:
Could you repurpose the no-allocation code that you introduced in the AspNetCore project here as well?

Reply:
Moved all the header code to `PrometheusHeadersParser` using the no-alloc code and shared it between both projects